encrypted-config-value/src/main/java/com/palantir/config/crypto/algorithm/Algorithm.java

```diff
@@ -30,28 +29,35 @@
  * algorithm with a supported key.
  */
 public enum Algorithm {
-    AES("AES", () -> AesKeyPair.newKeyPair(), () -> AesGcmEncrypter.INSTANCE),
-    RSA("RSA", () -> RsaKeyPair.newKeyPair(), () -> RsaOaepEncrypter.INSTANCE);
+    AES("AES", AesGcmEncrypter.INSTANCE) {
+        @Override
+        public KeyPair newKeyPair() {
```

Comment:
errorprone demands this?

https://github.com/google/error-prone/blob/master/docs/bugpattern/ImmutableEnumChecker.md

Reply:
ok so we were being naughty with our enums before by having them take suppliers I guess